Assets/Scripts/DataController.cs

```diff
@@ -372,12 +346,36 @@ public void ChangeStarSelection(GameObject selectedStar)
         }
     }
 
-    public void UpdateMagnitudeThreshold(float newVal)
+    public void SetMagnitudeThreshold(float newVal)
     {
         magnitudeThreshold = newVal;
-        foreach (GameObject starObject in GameObject.FindGameObjectsWithTag("Star"))
+        for(int i = 0; i < allStarComponents.Length; i++)
         {
-            showStar(starObject, starObject.GetComponent<StarComponent>().starData.Mag < magnitudeThreshold);
+            showStar(allStarComponents[i].gameObject, allStarComponents[i].gameObject.GetComponent<StarComponent>().starData.Mag < magnitudeThreshold);
```

Comment:
allStarComponents[i] is a StarComponent, so it's redundant to go get its GameObject then go back and find the same component you started with :)

Reply:
Thanks for catching that!  Fixed!